docs(protocol): add comment for EIP-1559 (#16389)
taikoxyz/taiko-mono: packages/protocol/docs/eip1559_on_l2.md

## Adjust the slope

To adjust the slope of the curve to satisfy $R == basefee(2T)/basefee(T)$, we simply need to choose $M$ and $b_0$. $b_0$ is simply to decide -- if we believe the cost of a L2 transaction is $1/n$ of the same L1 transaction, we simply use the current L1 base fee divided by $n$. Then we can simply tune $M$ to make sure $R == basefee(2T)/basefee(T)$ holds. This is very simply manually a try-and-adjust approach as shown in `Lib1559Math.t.sol`. The TaikoL1 contract will check if $R == basefee(2T)/basefee(T)$ holds but will not calculate $M$ for us.

## Implementation Difference

> Our implementation computes the basefee prior to applying the gas used by the most recent block (indicated by the blue line in the image below), in contrast to Vitalik's concept, which calculates the average basefee using the gas consumption of the latest block (represented by the red line).

Reviewer: ```suggestion
Our implementation computes the basefee prior to applying the gas used by the current block (indicated by the blue line in the image below), in contrast to Vitalik's concept, which calculates the average basefee by also taking into account the gas consumption of the current block (represented by the red line).
```

Just changed the wording to use "current block" because "most recent' and "latest" could be interpreted in different ways.
